test/IntegrationTests.py

```diff
@@ -11,6 +11,36 @@
 
 from selenium import webdriver
 from selenium.webdriver.chrome.options import Options
+from selenium.webdriver.common.desired_capabilities import DesiredCapabilities
+
+
+class WebdriverLogFacade(object):
+
+    last_timestamp = 0
+
+    def __init__(self, webdriver):
+        self._webdriver = webdriver
+
+    def get_log(self):
+        last_timestamp = self.last_timestamp
+        entries = self._webdriver.get_log("browser")
+        filtered = []
+
+        for entry in entries:
+            # check the logged timestamp against the
+            # stored timestamp
+            if entry["timestamp"] > self.last_timestamp:
+                filtered.append(entry)
+
+                # save the last timestamp only if newer
+                # in this set of logs
+                if entry["timestamp"] > last_timestamp:
+                    last_timestamp = entry["timestamp"]
+
+        # store the very last timestamp
+        self.last_timestamp = last_timestamp
+
+        return filtered
```

Comment:
To be called once per test (at least once in the teardown phase) to get the test-related browser errors

Reply:
I really like this solution - much cleaner than the `window.tests` hack, and (cc @valentijnnieman) it will catch errors during page load as well, presumably.

I'm curious though, why a separate class, rather than just a method of `IntegrationTests`?

Reply:
No specific reason. Found it while looking for the capabilities finer points in Python and liked it. Did not stop to consider if it made more or less sense to have it standalone. I can imagine wanting to make this richer and keeping it in a class sort of marginally makes sense in that context.

Reply:
Looks good :-) I think it would make more sense to me to have it as a method if `IntegrationTests` instead of a separate class - just to keep everything in one place.

Reply:
Sure. I can move this into IntegrationTests